File: samples/fortran/FibonacciCalculator.f90
program FibonacciCalculator

  implicit none
  integer :: n, i
  integer, allocatable :: fibonacci(:)

  ! Get user input
  print *, 'How many terms of Fibonacci sequence would you like to display?'
  read *, n

  ! Validate user input
  if (n <= 0) then
     print *, 'Please enter a positive number.'
     stop
  end if

  ! Allocate memory for Fibonacci sequence
  allocate(fibonacci(n))

  ! Initialize first two terms
  if (n >= 1) fibonacci(1) = 0
  if (n >= 2) fibonacci(2) = 1

  ! Calculate Fibonacci sequence
  do i = 3, n
     fibonacci(i) = fibonacci(i-1) + fibonacci(i-2)
  end do

  ! Display Fibonacci sequence
  print *, 'The first ', n, ' terms of the Fibonacci sequence are:'
  print *, fibonacci

  ! Free the allocated memory
  deallocate(fibonacci)

end program FibonacciCalculator

File: samples/pas/fibonaccicalculator.pas
program FibonacciCalculator;

function Fibonacci(n: LongInt): Int64;
var
    a, b, tmp: Int64;
    i: LongInt;
begin
    if n <= 1 then
        Fibonacci := n
    else
    begin
        a := 0;
        b := 1;
        for i := 2 to n do
        begin
            tmp := a + b;
            a := b;
            b := tmp;
        end;
        Fibonacci := b;
    end;
end;

var
    num: LongInt;
begin
    Write('Enter the position of Fibonacci number you want to calculate: ');
    ReadLn(num);
    
    if num < 0 then
        WriteLn('Please enter a non-negative number.')
    else
        WriteLn('Fibonacci(', num, ') = ', Fibonacci(num));
    
    ReadLn;
end.
